target_postgres/connector.py

```diff
@@ -118,13 +118,24 @@ def prepare_table(  # type: ignore[override]
         table = meta.tables[
             full_table_name
         ]  # So we don't mess up the casing of the Table reference
+
+        columns = self.get_table_columns(
```

Comment:
Wow this was worse than I expected.

I thought our gain would be mainly from caching all of the table's and doing a single metadata query I had no idea we were doing a query for every column!